app/factories/bulkrax/valkyrie_object_factory.rb

```diff
@@ -264,25 +263,57 @@ def create_work(attrs)
       end
     end
 
-    def combined_files_with(remote_files:)
+    ## Prepare fileset data in the required format for creating or updating a work
+    # TODO: Determine why attrs is different from attributes?
+    # TODO: Disabled s3 until we get additional details
+    def prep_fileset_content(attrs)
+      # combine remote_files + thumbnail_url [Array < { url:, file_name:, * }]
       thumbnail_url = HashWithIndifferentAccess.new(self.attributes)['thumbnail_url']
+      all_remote_files = merge_thumbnails(remote_files: attrs["remote_files"], thumbnail_url: thumbnail_url)
+      # combine local & remote files [Array < Hash &/or String]
+      all_local_files = self.attributes['file']
+      all_files = all_local_files + all_remote_files
+
+      # collect all uploaded files [Array < Hyrax::UploadedFile]
+      uploaded_local = uploaded_local_files(uploaded_files: attrs[:uploaded_files])
```

Comment:
apologies if I'm missing it, but do we need a fallback here for when these fields don't exist? like: 

```
all_remote_files = merge_thumbnails(remote_files: attrs["remote_files"] || [], thumbnail_url: thumbnail_url)
uploaded_local = uploaded_local_files(uploaded_files: attrs[:uploaded_files] || [])
uploaded_remote = uploaded_remote_files(remote_files: all_remote_files)
```

Reply:
You're right, we aren't calling that method without the parameter, but if we did, it would break. I changed the default to an empty array since remote_files should always be an array of hashes.